Userland/Services/FileSystemAccessServer/ClientConnection.cpp

```diff
@@ -0,0 +1,125 @@
+/*
+ * Copyright (c) 2021, timmot <tiwwot@protonmail.com>
+ *
+ * SPDX-License-Identifier: BSD-2-Clause
+ */
+
+#include <AK/Debug.h>
+#include <FileSystemAccessServer/ClientConnection.h>
+#include <LibC/fcntl.h>
+#include <LibCore/File.h>
+#include <LibCore/StandardPaths.h>
+#include <LibGUI/Application.h>
+#include <LibGUI/FilePicker.h>
+
+namespace FileSystemAccessServer {
+
+static HashMap<int, RefPtr<ClientConnection>> s_connections;
+
+ClientConnection::ClientConnection(NonnullRefPtr<Core::LocalSocket> socket, int client_id)
+    : IPC::ClientConnection<FileSystemAccessClientEndpoint, FileSystemAccessServerEndpoint>(*this, move(socket), client_id)
+{
+    s_connections.set(client_id, *this);
+}
+
+ClientConnection::~ClientConnection()
+{
+}
+
+void ClientConnection::die()
+{
+    s_connections.remove(client_id());
+    exit(0);
+}
+
+Messages::FileSystemAccessServer::RequestFileReadResponse ClientConnection::request_file_read(String const& path)
+{
+    auto fd = open(path.characters(), O_RDONLY);
```

Comment:
Maybe we should check that the user provided an absolute path with a simple `path.starts_with("/")`, since relative paths would be relative to the CWD of FileSystemAccessServer, which would only lead to bugs. I don't think we should `VERIFY`, but somehow respond with an error code or error message.

Reply:
The request_{read,write} methods are more developer facing than the prompts so I think it would make sense to reject any relative paths outright, or we could have an argument for the base path from the client?